rs/crypto/ecdsa_secp256k1/src/lib.rs

```diff
@@ -466,19 +521,41 @@ impl PrivateKey {
             k256::schnorr::SigningKey::from(&self.key)
         };
 
+        bip340
+            .sign_prehash_with_aux_rand(message, aux_rand)
+            .map(|s| s.to_bytes())
+            .ok()
+    }
+
+    /// Sign a message with BIP340 Schnorr
+    pub fn sign_message_with_bip340<R: Rng + CryptoRng>(
+        &self,
+        message: &[u8; 32],
+        rng: &mut R,
+    ) -> [u8; 64] {
         loop {
             /*
              * The only way this function can fail is the (cryptographically unlikely)
              * situation where k or s of zero is generated. If this occurs, simply retry
              * with a new aux_rand
              */
             let aux_rand = rng.gen::<[u8; 32]>();
-            if let Ok(sig) = bip340.sign_prehash_with_aux_rand(message, &aux_rand) {
-                return sig.to_bytes();
+            if let Some(sig) = self.sign_bip340_with_aux_rand(message, &aux_rand) {
+                return sig;
             }
         }
     }
 
+    /// Sign a message with BIP340 Schnorr without using an external RNG
+    ///
+    /// The aux_rand parameter for BIP340 is just to re-randomize the signature,
```

Comment:
nit: This part of the documentation seems outdated and should probably be removed.